tests/TestCase/ORM/AssociationTest.php

```diff
@@ -330,6 +330,48 @@ public function testProperty()
     }
 
     /**
+     * Test that warning is shown if property name clashes with table field.
+     *
+     * @return void
+     * @expectedException PHPUnit_Framework_Error_Warning
+     */
+    public function testPropertyNameClash()
+    {
+        $this->source->schema(['foo' => ['type' => 'string']]);
+        $this->assertEquals('foo', $this->association->property());
```

Comment:
With the changes I made to avoid overriding of `Association::property()` in child classes I don't think having a test in each implementation is necessary.